src/data/nat/choose/multinomial.lean

```diff
@@ -0,0 +1,171 @@
+/-
+Copyright (c) 2022 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+Heavily inspired by code from Kyle Miller and Kevin Buzzard
+-/
+import algebra.big_operators.basic
+import algebra.big_operators.fin
+import data.nat.choose.basic
+import data.finset.basic
+import data.list.perm
+import data.fin.vec_notation
+
+import tactic.linarith
+
+/-!
+# Multinomial
+
+This file defines the multinomial coefficient and several small lemma's for manipulating it.
+
+## Main declarations
+
+- `nat.multinomial`: the multinomial coefficient
+
+-/
+
+open_locale nat
+
+namespace nat
+
+/-- The multinomial coefficient. Gives the number of strings consisting of symbols
+from `s`, where `c ∈ s` appears with multiplicity `f c`.
+
+Defined as `(s.sum f)! / s.prod (factorial ∘ f)`
+-/
+def multinomial {α} (s : finset α) (f : α → ℕ) : ℕ := (s.sum f)! / s.prod (factorial ∘ f)
+
+
+lemma prod_factorial_dvd_factorial_sum {α} (s : finset α) (f : α → ℕ) :
+  s.prod (factorial ∘ f) ∣ (s.sum f)! :=
+begin
+  classical,
+  induction s using finset.induction with α' s' has ih,
+  { simp only [finset.sum_empty, finset.prod_empty, factorial], },
+  { simp [finset.prod_insert has, finset.sum_insert has],
```

Comment:
Non-terminal simp needs to be squeezed (replace by output of `squeeze_simp [...]`).